### PR TITLE
added note on external volumes with docker stack deploy

### LINKED ISSUE #3141: "docker stack deploy" creates external volumes that do not exist, instead of exiting
File: [compose/compose-file/index.md](https://docs.docker.com/compose/compose-file/), CC @londoncalling 

This concerns the "external" property for volumes declared in the top-level "volumes" section of a STACK file.
https://docs.docker.com/compose/compose-file/#external

It appears that if a volume declared as "external" does NOT already exist, it is created using the "local" driver.  This behavior is inconsistent with the documentation:
screenshot of safari 5-2-17 4-33-31 pm ... and also with how `docker secrets` treats the external property.

```
$ docker volume ls
DRIVER              VOLUME NAME

$ cat dc-test-external-exits.yml
version: "3.1"
services:
  aservice:
    image: busybox
    volumes:
      - test-vol:/tmp:z
volumes:
  test-vol:
    external: true

$ docker stack deploy -c dc-test-external-exits.yml myapp
Creating network myapp_default
Creating service myapp_aservice

$ docker volume ls
DRIVER              VOLUME NAME
local               test-vol
```

Docker version info:
```
$ docker version
Client:
 Version:      17.03.1-ce
 API version:  1.27
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Tue Mar 28 00:40:02 2017
 OS/Arch:      darwin/amd64

Server:
 Version:      17.03.1-ce
 API version:  1.27 (minimum version 1.12)
 Go version:   go1.7.5
 Git commit:   c6d412e
 Built:        Fri Mar 24 00:00:50 2017
 OS/Arch:      linux/amd64
 Experimental: true
```

### PR DESCRIPTION
**Pull request by londoncalling**
### What's changed

- added note about the differences in behaviour between `docker stack deploy` and `docker compose up` with regard to external volumes that do not exist

- Note, this solves only the docs issue. Related issues are listed below.

### Related

Fixes #3141 (docs)

Moby: https://github.com/moby/moby/issues/29976
Moby: https://github.com/moby/moby/issues/33181 (duplicate, closed)

### Reviewers, fyi

@briantd @cghislai @dnephin @shin- @justincormack @thaJeztah 

I'm going to merge to get this published, as the docs issue has been sitting a while. Y'all please ping me if you want changes to the docs update on this PR. 

Signed-off-by: Victoria Bialas <victoria.bialas@docker.com>

### CODE REVIEW

**Comment by londoncalling** [comment]
LGTM

File: compose/compose-file/index.md [hunk 1]
```diff
@@ -1431,6 +1431,18 @@ refer to it within the Compose file:
         external:
           name: actual-name-of-volume
 
+> External volumes are always created with docker stack deploy
+>
+External volumes that do not exist _will be created_ if you use [docker stack
```
Comment (thaJeztah) [inline compose/compose-file/index.md]: Sorry for my late reply; I wonder if we should call out that this is not really "by design", but a current "limitation". Basically, it's a bug (or, indeed a current limitation); I think it's important to add "some" note about that. Users should not rely on this auto-creation, because (hopefully) we'll be able to fix it, and the behaviour will change in future.

TL;DR if you define a volume as _external_, you should create it up front.

but ping @cpuguy83 for input as well
